helpers/persistPostVisitActions.ts

```diff
@@ -175,6 +171,108 @@ const postNoteToBackend = async (
   }
 };
 
+export const persistUnableToEnterPostVisitActions = async (
+  unableToEnterData: Partial<
+    ProcessDatabaseSchema["schema"]["unableToEnter"]["value"]
+  >,
+  processRef: ProcessRef
+): Promise<void> => {
+  if (!unableToEnterData) {
+    return;
+  }
+
+  const unableToEnterMap = processPostVisitActionMap["unableToEnter"];
+  const externalDatabase = await Storage.ExternalContext?.database;
+
+  if (!externalDatabase) {
+    return;
+  }
+  const residents = await externalDatabase.get("residents", processRef);
+  const tenantNames = residents
+    ? residents.tenants.map((tenant) => tenant.fullName).join(", ")
+    : "N/A";
+  const address = residents ? residents.address.join(", ") : "N/A";
+
+  const firstFailedAttempt = unableToEnterData["firstFailedAttempt"];
+  const secondFailedAttempt = unableToEnterData["secondFailedAttempt"];
+  const thirdFailedAttempt = unableToEnterData["thirdFailedAttempt"];
+  const fourthFailedAttempt = unableToEnterData["fourthFailedAttempt"];
+
+  const firstFailedAttemptNotes = firstFailedAttempt
+    ? firstFailedAttempt["notes"] || "N/A"
+    : "N/A";
+  const secondFailedAttemptNotes = secondFailedAttempt
+    ? secondFailedAttempt["notes"] || "N/A"
+    : "N/A";
+  const thirdFailedAttemptNotes = thirdFailedAttempt
+    ? thirdFailedAttempt["notes"] || "N/A"
+    : "N/A";
+  const fourthFailedAttemptNotes = fourthFailedAttempt
+    ? fourthFailedAttempt["notes"] || "N/A"
+    : "N/A";
+
+  const firstAttemptDate = firstFailedAttempt
+    ? firstFailedAttempt["date"] || "N/A"
+    : "N/A";
+  const secondAttemptDate = secondFailedAttempt
+    ? secondFailedAttempt["date"] || "N/A"
+    : "N/A";
+  const thirdAttemptDate = thirdFailedAttempt
+    ? thirdFailedAttempt["date"] || "N/A"
+    : "N/A";
+  const fourthAttemptDate = fourthFailedAttempt
+    ? fourthFailedAttempt["date"] || "N/A"
+    : "N/A";
+
+  if (thirdFailedAttempt) {
+    if (thirdFailedAttempt.needsAppointmentLetterReminder) {
+      const map =
+        unableToEnterMap["thirdFailedAttempt.needsAppointmentLetterReminder"];
+      const description = `Action: Third Failed Attempt - Appointment Letter Reminder
+      Address: ${address}.
+      Resident(s) details: ${tenantNames}.
+      First Attempt: ${firstAttemptDate}, Notes: ${firstFailedAttemptNotes}.
+      Second Attempt: ${secondAttemptDate}, Notes: ${secondFailedAttemptNotes}.
+      Third Attempt: ${thirdAttemptDate}, Notes: ${thirdFailedAttemptNotes}`;
+
+      postValueToBackend(description, map, processRef);
+    }
+  }
+
+  if (fourthFailedAttempt) {
+    const description = `Address: ${address}.
```

Comment:
```suggestion
    const details = `Address: ${address}.
```

So you can create `description` inside, and make this code more uniform.